### PR TITLE
RE2022-177: refactor tool runner to include env

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: src/loaders/compute_tools/tool_common.py [hunk 1]
```diff
@@ -76,14 +76,13 @@ def __init__(
         -h, --help            show this help message and exit
 
         required named arguments:
+        --env {CI,NEXT,APPDEV,PROD,None}
+                              Environment used to run the tool.
+        --kbase_collection KBASE_COLLECTION
+                              KBase collection identifier name.
         --load_ver LOAD_VER   KBase load version (e.g. r207.kbase.1).
```
Comment (MrCreosote) [inline src/loaders/compute_tools/tool_common.py:83]: Hmm, looking at this - the tools are computing on data in `collectionssource/`, right? In which case this isn't a load_ver, it's a source ver.

File: src/loaders/compute_tools/tool_common.py [hunk 2]
```diff
@@ -148,21 +156,20 @@ def _parse_args(self):
 
         # Required flag arguments
         required.add_argument(
-            f'--{loader_common_names.LOAD_VER_ARG_NAME}', required=True, type=str,
-            help=loader_common_names.LOAD_VER_DESCR
+            f"--{loader_common_names.ENV_ARG_NAME}", required=True, type=str,
+            choices=loader_common_names.KB_ENV + [loader_common_names.DEFAULT_ENV],
```
Comment (MrCreosote) [inline src/loaders/compute_tools/tool_common.py:160]: Should this default to PROD like the workspace downloader or be required, do you think? The former would be less work for users but easier to accidentally use the wrong environment so I'm not really sure.

File: src/loaders/compute_tools/tool_common.py [hunk 3]
```diff
@@ -148,21 +156,20 @@ def _parse_args(self):
...
+            help=f"Environment used to run the tool."
```
Comment (MrCreosote) [inline src/loaders/compute_tools/tool_common.py:161]:
```suggestion
            help=f"Environment containing the data to be processed."
```

File: src/loaders/compute_tools/tool_common.py (hunk 1 above)
Comment (MrCreosote) [reply src/loaders/compute_tools/tool_common.py:83]: yeah, I see we're using load ver in both the input and output dir, which doesn't really work - we may recompute data from the same source version leading to a new load_ver

Comment (Tianhao-Gu) [reply src/loaders/compute_tools/tool_common.py]: 👍 added `source_ver` and defaults `load_ver` to `source_ver`

File: src/loaders/compute_tools/tool_common.py (hunk 2 above)
Comment (Tianhao-Gu) [reply src/loaders/compute_tools/tool_common.py]: 👍  making PROD as default.

File: src/loaders/compute_tools/tool_common.py (hunk 3 above)
Comment (Tianhao-Gu) [reply src/loaders/compute_tools/tool_common.py]: 👍

File: src/loaders/compute_tools/entrypoint.sh [hunk 4]
```diff
@@ -7,6 +7,7 @@ echo "using $env_tool to manage environment"
 $env_tool run -n $CONDA_ENV \
   python $PY_SCRIPT \
   --load_ver $LOAD_VER \
+  --source_ver $SOURCE_VER \
```
Comment (MrCreosote) [inline src/loaders/compute_tools/entrypoint.sh:10]: I guess we're going to have to rebuild & push the 3 tool images yeah?

File: src/loaders/compute_tools/tool_common.py [hunk 5]
```diff
@@ -73,33 +73,37 @@ def __init__(
...
-        --env {CI,NEXT,APPDEV,PROD,None}
-                              Environment used to run the tool.
-        --kbase_collection KBASE_COLLECTION
-                              KBase collection identifier name.
-        --load_ver LOAD_VER   KBase load version (e.g. r207.kbase.1).
+          --kbase_collection KBASE_COLLECTION
+                                KBase collection identifier name.
+          --source_ver SOURCE_VER
+                                Version of the source data. (e.g. 207, 214 for GTDB,
+                                2023.06 for GROW/PMI)
```
Comment (MrCreosote) [inline src/loaders/compute_tools/tool_common.py:83]: Maybe mention that it needs to match the dir in `collectionssource`?

**Review by MrCreosote — APPROVED** [review]
LGTM

File: src/loaders/compute_tools/tool_common.py (hunk 3 above)
Comment (MrCreosote) [reply src/loaders/compute_tools/tool_common.py]: :+1:

File: src/loaders/compute_tools/tool_common.py (hunk 2 above)
Comment (MrCreosote) [reply src/loaders/compute_tools/tool_common.py]: :+1:

File: src/loaders/compute_tools/tool_common.py (hunk 1 above)
Comment (MrCreosote) [reply src/loaders/compute_tools/tool_common.py]: :+1:

File: src/loaders/common/loader_common_names.py [hunk 6]
```diff
@@ -16,12 +16,17 @@
 KBASE_COLLECTION_ARG_NAME = "kbase_collection"
 # Description of the --kbase_collection argument in various loaders programs.
 KBASE_COLLECTION_DESCR = (
-    f"KBase collection identifier name (default: {DEFAULT_KBASE_COLL_NAME})."
```
Comment (MrCreosote) [inline src/loaders/common/loader_common_names.py:19]: Missed one thing - is this going to affect other CLIs?

File: src/loaders/compute_tools/entrypoint.sh (hunk 4 above)
Comment (Tianhao-Gu) [reply src/loaders/compute_tools/entrypoint.sh:10]: yup.

File: src/loaders/compute_tools/tool_common.py (hunk 5 above)
Comment (Tianhao-Gu) [reply src/loaders/compute_tools/tool_common.py]: 👍

File: src/loaders/common/loader_common_names.py (hunk 6 above)
Comment (Tianhao-Gu) [reply src/loaders/common/loader_common_names.py:19]: I think we are moving away from having `GTDB` or any others as default collection. Currently, only the  taxa_count and gtdb_genome_attribs_load has default kbase_collection as `GTDB`. I think for a description, that's fine.

Comment (MrCreosote) [reply src/loaders/common/loader_common_names.py:19]: :+1:

File: src/loaders/compute_tools/tool_common.py (hunk 5 above)
Comment (MrCreosote) [reply src/loaders/compute_tools/tool_common.py]: :+1:

File: src/loaders/compute_tools/tool_common.py [hunk 7]
```diff
@@ -148,21 +165,28 @@ def _parse_args(self):
...
-            help=loader_common_names.LOAD_VER_DESCR
+            f'--{loader_common_names.KBASE_COLLECTION_ARG_NAME}', required=True, type=str,
+            help=loader_common_names.KBASE_COLLECTION_DESCR
         )
         required.add_argument(
-            '--source_data_dir', required=True, type=str,
-            help='Source data (e.g. genome files) directory. '
-                 + '(e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207'
+            f'--{loader_common_names.SOURCE_VER_ARG_NAME}', required=True, type=str,
+            help="Version of the source data, which should match the source directory in the collectionssource. (e.g. 207, 214 for GTDB, 2023.06 for GROW/PMI)"
```
Comment (MrCreosote) [inline src/loaders/compute_tools/tool_common.py:173]: line wrap, but that can be fixed in a later PR AFAIC

**Review by MrCreosote — APPROVED** [review]
LGTM

File: src/loaders/compute_tools/tool_common.py (hunk 7 above)
Comment (Tianhao-Gu) [reply src/loaders/compute_tools/tool_common.py:173]: 👍  I will fix it in the upcoming PR.

**Pull request merged**